--- a/dng_sdk/source/dng_jxl.cpp
+++ b/dng_sdk/source/dng_jxl.cpp
@@ -1537,6 +1537,7 @@
 			
 		// XMP.
 
+		#if qDNGUseXMP
 		if (includeXMP && metadata && metadata->GetXMP ())
 			{
 
@@ -1574,6 +1575,7 @@
 				}
 
 			} // xmp
+		#endif
 
 		// IPTC.
 
@@ -3808,6 +3810,7 @@
 void dng_jxl_decoder::ProcessXMPBox (dng_host &host,
 									 const std::vector<uint8> &data)
 	{
+	#if qDNGUseXMP
 	
 	#if qDNGValidate
 
@@ -3873,6 +3876,7 @@
 			}
 
 		}
+	#endif
 	
 	}
 
